### PR TITLE
Support deploying Elasticsearch Curator

### PR DESCRIPTION
**Pull request by dougszumski**
This allows users to supply an Elasticsearch Curator actions file
to manage log retention [1]. Curator then runs on a cron job, which
defaults to 1 day. No attempt is made to provide a default retention
policy because some sites may not want one at all.

[1] https://www.elastic.co/guide/en/elasticsearch/client/curator/current/actionfile.html

### CODE REVIEW

File: ansible/roles/elasticsearch/defaults/main.yml [hunk 1]
```diff
@@ -21,11 +31,31 @@ elasticsearch_cluster_name: "kolla_logging"
...
 
+
+#######################
+# Elasticsearch Curator
+#######################
+
+# How frequently Curator runs. On multinode deployments of Curator
+# you may wish to override this in hostvars so that Curator does
+# not run simultaneously on all nodes.
+elasticsearch_curator_cron_table: "0 0 * * *"
```
Comment (JohnGarbutt) [inline ansible/roles/elasticsearch/defaults/main.yml]: Might be nice to say what frequency this default maps to?

**Comment by JohnGarbutt** [comment]
I actually think we should be more opinonated, and just have a kolla variable to set the retention period. If you don't want that, simply turn curator off.

Does that sound crazy?

File: ansible/group_vars/all.yml [hunk 2]
```diff
@@ -581,6 +581,8 @@ monasca_control_plane_project: "monasca_control_plane"
 
 elasticsearch_address: "{{ kolla_internal_fqdn }}"
 enable_elasticsearch: "{{ 'yes' if enable_central_logging | bool or enable_freezer | bool or enable_osprofiler | bool or enable_skydive | bool or enable_monasca | bool else 'no' }}"
+enable_elasticsearch_curator: "no"
```
Comment (JohnGarbutt) [inline ansible/group_vars/all.yml]: yes if enable_elasticsearch?

Comment (JohnGarbutt) [reply ansible/group_vars/all.yml]: Actually, I see why it should be off by default.

But when we turn it on, I think it should do something by default.

**Comment by JohnGarbutt** [comment]
There is a nice example here, for logstash:
https://community.talend.com/t5/Architecture-Best-Practices-and/Deleting-old-indices-in-Elasticsearch-using-Curator/ta-p/124471

Prometheus defaults to two week retention, although that is expected I guess.

What about this for a default... close the index after 30 days (i.e. free up the RAM, stops you being able to search), and delete the index after 60 days (i.e. free up the disk space later)?

That way if you notice stuff go missing, you have time to work out how you can get it back.

But off by default makes sense.

File: releasenotes/notes/add-elasticsearch-curator-88089d04f7ccd549.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Add Elasticsearch Curator for managing aggregated log data.
```
Comment (JohnGarbutt) [inline releasenotes/notes/add-elasticsearch-curator-88089d04f7ccd549.yaml]: More for upstream, we should probably say how to turn it on, and what it does if you turn it on.

**Comment by dougszumski** [comment]
> Nit: can you add the Change-Id from https://review.opendev.org/#/c/694537/ which will make it easier to identify later.

No problem, done

**Pull request merged**